silverback/application.py

```diff
@@ -88,6 +107,92 @@ def __init__(self, settings: Settings | None = None):
             f"{signer_str}{start_block_str}{new_block_timeout_str}"
         )
 
+        # NOTE: The runner needs to know the set of things that the app is tracking as a parameter
+        # NOTE: We also need to know the defaults in case the parameters are not in the backup
+        self.parameter_defaults: dict[str, ScalarType | None] = dict()
+
+        self._create_system_startup_task()
+        self._create_system_backup_task()
+
+    def _create_system_startup_task(self):
+        # Add a task to load all parameters from state at startup
+        async def startup_handler() -> StateSnapshot:
+            # NOTE: This is not networked in any way, nor thread-safe nor multi-process safe,
+            #       but will be accessible across multiple workers in a single container
+            # NOTE: *DO NOT USE* in Runner, as it will not be updated by the app
+            self.state = SharedState()
```

Comment:
question: I thought `self.state` was runner only?

question: Does this really set `self.state` for *all* workers?  Because this job will only get processed by one.

Reply:
> question: I thought `self.state` was runner only?

I am going to remove runner state

> question: Does this really set `self.state` for _all_ workers? Because this job will only get processed by one.

Yes it does

Reply:
> Yes it does. Currently how it works is that app.state is not accessible until this task is executed, and then all workers have access to it

You're sure about that?  Kind of looks like taskiq creates separate processes for workers using the `multiprocessing` module.

Reply:
```bash
$ SILVERBACK_BROKER_CLASS="taskiq_redis:ListQueueBroker" SILVERBACK_BROKER_URI="redis://localhost:6379" silverback worker -w 3 example:app --network :mainnet
SUCCESS: Loaded Silverback App:
  NETWORK="ethereum:mainnet"
  SIGNER=None
INFO: Found cached value for app.state['system:last_block_seen']: 19815550
INFO: Found cached value for app.state['system:last_block_processed']: 19815550
INFO: Found cached value for app.state['bad_number']: 3 <--- Set to 3 here on startup
SUCCESS: app_startup - 0.000s (0.0%)
...  # very many block and event log triggers
SUCCESS: Update: app.state['bad_number'] = 4 <--- Updates to 4 here
... # some more block and event log triggers
Exception found while executing function: I don't like the number 4.
Traceback (most recent call last):
  ...
  File "~/work/Silverback/sdk/example.py", line 78, in exec_event1
    raise ValueError(f"I don't like the number {app.state.bad_number}.")
ValueError: I don't like the number 4.
```

Reply:
I mean, there is maybe the slight chance here that it does not randomly distribute to the workers and is always using the same one to trigger each task such that I don't see the issue, but I've been playing with this for a day and half and it seems to work

Reply:
I added a 5 second delay (such that it should pick a different worker) and a print statement:
```
SUCCESS: Loaded Silverback App:
  NETWORK="ethereum:mainnet"
  SIGNER=None
INFO: Found cached value for app.state['system:last_block_seen']: 19815731
INFO: Found cached value for app.state['system:last_block_processed']: 19815731
INFO: Found cached value for app.state['bad_number']: 4
SUCCESS: app_startup - 0.000s (0.0%)
INFO: `ape-cache` database has not been initialized
SUCCESS: exec_block[block_number=19815738,block_hash=0xb691ca40a760fb6eb3c883b5c4beec4263ee537db69493517e7aaad7ad8665c6] - 0.420s (3.2%)
4
SUCCESS: exec_event1[contract_address=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,event_signature="Transfer(address indexed sender, address indexed receiver, uint256 amount)",block_number=19815738,transaction_hash=0xf76096051871d479cdeba6c408cb5ac3f5f880ac4e02185c1a73387962ef3fb0,log_index=28] - 5.010s (38.5%)
4
SUCCESS: exec_event1[contract_address=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,event_signature="Transfer(address indexed sender, address indexed receiver, uint256 amount)",block_number=19815738,transaction_hash=0xf76096051871d479cdeba6c408cb5ac3f5f880ac4e02185c1a73387962ef3fb0,log_index=30] - 5.000s (38.5%)
SUCCESS: exec_block[block_number=19815739,block_hash=0x19ae3d9afd6e1e2a65da136764755be44233036526dbaf57590b04cbb88f520e] - 0.150s (1.2%)
4
SUCCESS: exec_event1[contract_address=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,event_signature="Transfer(address indexed sender, address indexed receiver, uint256 amount)",block_number=19815738,transaction_hash=0xf76096051871d479cdeba6c408cb5ac3f5f880ac4e02185c1a73387962ef3fb0,log_index=31] - 5.000s (38.5%)
SUCCESS: Update: app.state['bad_number'] = 3
3
SUCCESS: exec_event1[contract_address=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,event_signature="Transfer(address indexed sender, address indexed receiver, uint256 amount)",block_number=19815738,transaction_hash=0xf76096051871d479cdeba6c408cb5ac3f5f880ac4e02185c1a73387962ef3fb0,log_index=33] - 5.000s (38.5%)
SUCCESS: exec_block[block_number=19815740,block_hash=0x4a4239cf41b398179786106a500da6e0bde606a1f85df489e9ef55e8b79df3e6] - 0.150s (1.2%)
3
SUCCESS: exec_event1[contract_address=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,event_signature="Transfer(address indexed sender, address indexed receiver, uint256 amount)",block_number=19815738,transaction_hash=0x68f0a77d557131181800f6f1a66122d9f3a4816a4fc0a173e2a09724e6cbc758,log_index=67] - 5.000s (38.5%)
SUCCESS: app_shutdown - 0.000s (0.0%)
3
SUCCESS: exec_event1[contract_address=0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48,event_signature="Transfer(address indexed sender, address indexed receiver, uint256 amount)",block_number=19815738,transaction_hash=0x68f0a77d557131181800f6f1a66122d9f3a4816a4fc0a173e2a09724e6cbc758,log_index=70] - 5.000s (38.5%)
```

Reply:
I'm curious what you see if you dump `os.getpid()` and `os.getppid()`.  Otherwise, any other way to ensure they're running on separate workers?  

Feel like I'm missing something here.

Reply:
Yep, that intuition is correct, they are all running in the same process ID despite `-w 3` given to `silverback worker`